addon/background.js

```diff
@@ -58,3 +74,12 @@ browser.runtime.onInstalled.addListener((details) => {
     browser.conversations.invalidateCache().catch(console.error);
   }
 });
+
+// Request handler for getting contact details.
+// Accessible through browser._background.request({ type: "contactDetails", payload: contact })
+requestHandlers.push(async (msg) => {
+  if (msg.type !== "contactDetails") {
+    return null;
+  }
+  return contactManager.getContactFromNameAndEmail(msg.payload);
```

Comment:
I just had another thought - do we need to think about what happens if the conversation changes before this resolves?

I guess it would just get dropped maybe, or do we end up blocking things?

Reply:
It would still get executed, but shouldn't impact anything other than the caching...